### DISCUSSION TITLE
btrfs missing from current AMI

### DISCUSSION THREAD

**Discussion by gilfreund** [amazonlinux/amazon-linux-2022#251] [Q&A] [answered]
I noticed that btrfs is missing from current AL2022 image. From the docs is sees its [removed](https://docs.aws.amazon.com/linux/al2022/release-notes/removed-packages-al2022.html). 
Another user noticed it [available on the ECS image](https://github.com/awslabs/amazon-ebs-autoscale/issues/60#issuecomment-1370321364). 
What is to be expected in the release? There are users and projects that make active [used of it](https://github.com/awslabs/amazon-ebs-autoscale).

**Comment by danie-dejager** [comment]
on AL2022 there is support for `xfs` and `vfat`

```
→ cat /proc/filesystems
nodev   sysfs
nodev   tmpfs
nodev   bdev
nodev   proc
nodev   cgroup
nodev   cgroup2
nodev   cpuset
nodev   devtmpfs
nodev   debugfs
nodev   tracefs
nodev   securityfs
nodev   sockfs
nodev   bpf
nodev   pipefs
nodev   ramfs
nodev   hugetlbfs
nodev   devpts
nodev   autofs
        xfs
        zonefs
nodev   mqueue
nodev   selinuxfs
nodev   pstore
nodev   configfs
        fuseblk
nodev   fuse
nodev   fusectl
        vfat
nodev   rpc_pipefs
nodev   overlay
nodev   binfmt_misc
```
AMZN2 only supports `xfs` and `squashfs` by default.

```
→ cat /proc/filesystems
nodev   sysfs
nodev   tmpfs
nodev   bdev
nodev   proc
nodev   cgroup
nodev   cgroup2
nodev   cpuset
nodev   devtmpfs
nodev   debugfs
nodev   tracefs
nodev   securityfs
nodev   sockfs
nodev   bpf
nodev   pipefs
nodev   ramfs
nodev   hugetlbfs
nodev   devpts
nodev   autofs
        xfs
        zonefs
nodev   mqueue
nodev   pstore
        squashfs
nodev   rpc_pipefs
nodev   overlay
nodev   binfmt_misc
```

**Reply by gilfreund** [reply]
I can  (and do in some cases) is MDADM and xfs, but was in the process of moving to btrfs, which is available on most Linux distributions. I find it a strange decision, all though I see RHEL is going the same path.

**Reply by danie-dejager** [reply]
ZFS appears to be more performant than btrfs. Have you given a basic ZFS partition a spin instead?
https://www.youtube.com/watch?v=G785-kxFH_M

**Comment by stewartsmith** [comment, marked as answer]
We do not at this time have any plans on adding btrfs support to Amazon Linux.
